keyvault/client/src/main/java/com/azure/keyvault/SecretAsyncClient.java

```diff
@@ -0,0 +1,434 @@
+import com.azure.common.http.HttpPipeline;
+import com.azure.common.http.rest.PagedResponse;
+import com.azure.common.http.rest.Response;
+import com.azure.common.http.rest.SimpleResponse;
+import com.azure.common.http.rest.VoidResponse;
+import com.azure.common.implementation.RestProxy;
+import com.azure.keyvault.implementation.SecretAttributesPage;
+import com.azure.keyvault.models.DeletedSecret;
+import com.azure.keyvault.models.Secret;
+import com.azure.keyvault.models.SecretAttributes;
+import org.apache.commons.lang3.Validate;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Objects;
+import java.util.function.Function;
+
+public final class SecretAsyncClient extends ServiceClient {
+    static final String API_VERSION = "7.0";
+    static final String ACCEPT_LANGUAGE = "en-US";
+    static final int DEFAULT_MAX_PAGE_RESULTS = 25;
+    static final String CONTENT_TYPE_HEADER_VALUE = "application/json";
+
+    private String vaultEndpoint;
+    private final SecretService service;
+
+    /**
+     * Creates a SecretAsyncClient that uses {@code pipeline} to
+     * service requests
+     *
+     * @param vaultEndpoint URL for the Azure KeyVault service.
+     * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
+     */
+    SecretAsyncClient(URL vaultEndpoint, HttpPipeline pipeline) {
+        super(pipeline);
+        this.vaultEndpoint = vaultEndpoint.toString();
+        this.service = RestProxy.create(SecretService.class, this);
+    }
+
+    /**
+     * Creates a builder that can configure options for the SecretAsyncClient before creating an instance of it.
+     * @return A new Builder to create a SecretAsyncClient from.
+     */
+    public static SecretAsyncClientBuilder builder() {
+        return new SecretAsyncClientBuilder();
+    }
+
+
+
+    /**
+     * The set operation adds a secret to the Azure Key Vault.
+     * If the named secret already exists, Azure Key Vault creates a new version of that secret.
+     * This operation requires the secrets/set permission.
+     *
+     * <p> The {@code secret} is required along with its non-null fields secret.name and secret.value. The secret.expires,
+     * secret.contentType and secret.notBefore values in {@code secret} are optional. If not specified, no values are set
+     * for the fields. The secret.enabled field is set to true by Azure Key Vault, if not specified.</p>
+     *
+     * @param secret The Secret object containing information about the secret and its properties. The properties secret.name and secret.value must be non null.
+     * @throws NullPointerException if {@code secret} is {@code null}.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value()} contains the created {@link Secret}.
+     */
+    public Mono<Response<Secret>> setSecretAsync(Secret secret) {
```

Comment:
Current spec wording says to not have 'Async' in method names.

Reply:
Updated.